libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SalesforceWebViewCookieManager.swift

```diff
@@ -0,0 +1,118 @@
+        setCookieValue(cookieStore: cookieStore, cookieType: Self.CLIENT_SRC, domain: mainDomain, name: Self.CLIENT_SRC, value: clientSrc)
+        setCookieValue(cookieStore: cookieStore, cookieType: Self.SID_CLIENT, domain: mainDomain, name: Self.SID_CLIENT, value: sidClient)
+        setCookieValue(cookieStore: cookieStore, cookieType: Self.ORG_ID, domain: mainDomain, name: Self.ORG_ID, value: orgId)
+        setCookieValue(cookieStore: cookieStore, cookieType: Self.csrfTokenCookieName, domain: mainDomain, name: Self.csrfTokenCookieName, value: csrfToken)
+        
+        // Lightning domain cookies
+        setCookieValue(cookieStore: cookieStore, cookieType: "sid for lightning", domain: lightningDomain, name: sidCookieName, value: lightningSid)
+        setCookieValue(cookieStore: cookieStore, cookieType: Self.csrfTokenCookieName, domain: lightningDomain, name: Self.csrfTokenCookieName, value: csrfToken)
+        
+        // Content domain cookies
+        setCookieValue(cookieStore: cookieStore, cookieType: "sid for content", domain: contentDomain, name: sidCookieName, value: contentSid)
+        
+        // Vf domain cookies
+        setCookieValue(cookieStore: cookieStore, cookieType: "sid for vf", domain: vfDomain, name: sidCookieName, value: vfSid)
+        setCookieValue(cookieStore: cookieStore, cookieType: Self.CLIENT_SRC, domain: vfDomain, name: Self.CLIENT_SRC, value: clientSrc)
+        setCookieValue(cookieStore: cookieStore, cookieType: Self.SID_CLIENT, domain: vfDomain, name: Self.SID_CLIENT, value: sidClient)
+        setCookieValue(cookieStore: cookieStore, cookieType: Self.ORG_ID, domain: vfDomain, name: Self.ORG_ID, value: orgId)
+        
+        SFSDKHybridLogger.i(Self.self, message: "[\(Self.self) \(#function)]: done setting cookies for \(String(describing: userAccount.credentials.userId)).")
+    }
+
+    @MainActor private func setCookieValue(cookieStore: WKHTTPCookieStore, cookieType: String, domain: String?, name: String?, value: String?) {
+        guard let domain = domain, let name = name, let value = value else {
+            SFSDKHybridLogger.w(Self.self, message: "[\(Self.self) \(#function)]: unable to set \(cookieType) in domain:\(domain ?? "nil")")
+            return
+        }
+
+        let properties: [HTTPCookiePropertyKey: Any] = [
+            .name: name,
+            .value: value,
+            .domain: domain,
+            .path: "/",
+            .secure: true,
+            .discard: true
+        ]
+
+        if let cookie = HTTPCookie(properties: properties) {
+            cookieStore.setCookie(cookie)
+            SFSDKHybridLogger.d(Self.self, message: "[\(Self.self) \(#function)]: setting \(cookieType) in domain:\(domain) with value:\(value)")
+        }
+    }
+    
+    private func getHttpsUrlFromDomain(domain: String) -> URL {
+        if domain.contains("://") {
+            return URL(string: domain) ?? URL(string: "https://\(domain)")!
```

Comment:
I was wondering how this was used and if the force-unwrap could be avoided, but I don't see where this is called from?

Reply:
It's not needed on iOS. It came over when I converted the code over from kotlin (see https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/dev/libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/ui/SalesforceWebViewCookieManager.kt#L90).

Reply:
Planning to clean up logging / add tests today.